website/registrations/utils.py

```diff
@@ -0,0 +1,349 @@
+
+    def get_parsed(self):
+        parsed = []
+        for row in self.rows:
+            parsed.append({'csv_raw': row.get_raw_value(), 'csv_parsed': row.get_parsed_value()})
+        return {'schema_id': self.schema_id, 'registrations': parsed}
+
+    def validate(self):
+        self.validate_csv_header_list()
+        for row in self.rows:
+            row.validate()
+
+class Row():
+    def __init__(self, row_dict, validations, log_error):
+        self.row_dict = row_dict
+        self.cells = [Cell(header.lower(),
+                           value, validations[header.lower()],
+                           functools.partial(log_error, column_index=column_index))
+                           for column_index, (header, value) in enumerate(row_dict.items())]
+
+    def to_json(self):
+        json_value = {}
+        for cell in self.cells:
+            json_value.update(cell.get_parsed_value())
+        return json.dumps(json_value)
+
+    def get_metadata(self):
+        parsed_metadata = {}
+        for cell in self.cells:
+            if cell.is_metadata:
+                parsed_metadata.update(cell.get_parsed_value())
+        return parsed_metadata
+
+    def get_registration_responses(self):
+        parsed_responses = {}
+        for cell in self.cells:
+            if not cell.is_metadata:
+                parsed_responses.update(cell.get_parsed_value())
+        return parsed_responses
+
+    def get_parsed_value(self):
+        return {'metadata': self.get_metadata(),
+                'registration_responses': self.get_registration_responses()}
+
+    def get_raw_value(self):
+        raw_value = []
+        for cell in self.cells:
+            raw_value.append(cell.get_raw_value())
+        return ','.join(raw_value)
+
+    def validate(self):
+        for cell in self.cells:
+            cell.validate()
+
+class Cell():
+    @property
+    def is_metadata(self):
+        return self.header in METADATA_FIELDS.keys()
+
+    def __init__(self, header, value, validations, log_error):
+        self.header = header
+        self.value = value
+        self.validations = validations
+        self.field = Cell.field_instance_for(self.header.lower(),
+                                             value=self.value,
+                                             validations=self.validations,
+                                             log_error=functools.partial(log_error, header=self.header))
+
+    def validate(self):
+        self.field.parse()
+
+    def get_parsed_value(self):
+        return {self.header: self.field.parse()}
+
+    def get_raw_value(self):
+        return self.value
+
+    def to_json(self):
+        return json.dumps({self.header: self.field.parse()})
+
+    @classmethod
+    def field_instance_for(cls, name, **kwargs):
+        field_instance = None
+        if name in METADATA_FIELDS.keys():
+            if name in CONTRIBUTOR_METADATA_FIELDS:
+                field_instance = ContributorField(**kwargs)
```

Comment:
If we change `value`, `validations` and `log_error` into position arguments, then instead of doing `Field(**kwargs)` we can do `Field(*args)` here.

Reply:
is there any inherent issue to using `kwargs`? is it anti-pattern?

Reply:
See comment above. I believe kwargs imply that these arguments are optional, and constructors shouldn't take `value`, `validations` and `log_error` as optional args.